tests/foreman/api/test_subscription.py

```diff
@@ -282,23 +283,22 @@ def test_positive_candlepin_events_processed_by_STOMP(self):
             environment=entities.LifecycleEnvironment(id=org.library.id),
             auto_attach=True,
         ).create()
-        with VirtualMachine(distro=DISTRO_RHEL7) as vm:
-            vm.install_katello_ca()
-            vm.register_contenthost(org.name, ak.name)
-            host = entities.Host().search(query={'search': f'name={vm.hostname}'})
-            host_id = host[0].id
-            host_content = entities.Host(id=host_id).read_json()
-            assert host_content["subscription_status"] == 2
-            with manifests.clone() as manifest:
-                upload_manifest(org.id, manifest.content)
-            subscription = entities.Subscription(organization=org).search(
-                query={'search': f'name="{DEFAULT_SUBSCRIPTION_NAME}"'}
-            )[0]
-            entities.HostSubscription(host=host_id).add_subscriptions(
-                data={'subscriptions': [{'id': subscription.cp_id, 'quantity': 1}]}
-            )
-            host_content = entities.Host(id=host_id).read_json()
-            assert host_content["subscription_status"] == 0
-            response = entities.Ping().search_json()["services"]["candlepin_events"]
-            assert response["status"] == "ok"
-            assert "0 Failed" in response["message"]
+        self.content_host.install_katello_ca()
```

Comment:
Can't we continue using context manager here?

It's not clear at all what `self.content_host` is and where it comes from. There are multiple other tests in the same class that still use `VirtualMachine`, making it unclear why fixture introduces vm at class level to begin with.

I get it's not really a point of this change and you are not particularly concerned about pytest migration here. Perhaps usefulness of broker could be better demonstrated on test that already migrated.

Reply:
Unfortunately, this is due to a limitation in injecting into unittest class methods. The context manager approach would absolutely be valid, but I wanted to demonstrate a fixture-based approach. If you have a good candidate for an already converted test, I can try adding that to this pull request as well.